### ISSUE TITLE
Zactuator  "undefined reference to `MQTTtoONOFF"

### ISSUE THREAD

**Issue by Djang0Pepper** [1technophile/OpenMQTTGateway#556]
I've previoulsy tested other feature DHT,DS1820 working so easy !!
but with Zactuator :(
I've only decommented in User_config.h this line :
#define ZactuatorONOFF "ONOFF"    //ESP8266, Arduino, ESP32,  Sonoff RF Bridge

And I get 

_> ketch\main.ino.cpp.o: In function `MQTTtoSYS(char*, ArduinoJson::JsonObject&)':
> 
> C:\Users\fr3d\Documents\PlatformIO\Projects\OpenMQTTGateway_new\OpenMQTTGateway\main/main.ino:303: undefined reference to `MQTTtoONOFF(char*, char*)'
> 
> sketch\main.ino.cpp.o: In function `variantAsInteger<long unsigned int>':
> 
> C:\Users\fr3d\Documents\PlatformIO\Projects\OpenMQTTGateway_new\OpenMQTTGateway\main/main.ino:303: undefined reference to `MQTTtoONOFF(char*, char*)'
> 
> collect2.exe: error: ld returned 1 exit status_

Re-tested on a freshen redownloaded.

**Comment by Djang0Pepper** [comment]
solution :
#ifdef ZactuatorONOFF // **outside the jsonpublishing** macro due to the fact that we need to use simplepublishing with HA discovery
      MQTTtoONOFF(topicOri, jsondata);
      #endif      
    #endif
If I put it inside i can compile.

**Comment by 1technophile** [comment]
Hi,

Strange, I've tried to uncomment `#define ZactuatorONOFF ` in User_config.h but I did not get any build error.
May you give me the board used?

**Comment by thomas-merzenich** [comment]
I have the same error:
`/home/thomas/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: /tmp/arduino_build_562085/sketch/main.ino.cpp.o:/home/thomas/Arduino/OpenMQTTGateway-development/main/main.ino:1323: undefined reference to MQTTtoONOFF(char*, char*)`

`/home/thomas/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: /tmp/arduino_build_562085/sketch/main.ino.cpp.o: in function receivingMQTT(char*, char*):`

`/home/thomas/Arduino/libraries/ArduinoJson/src/ArduinoJson/JsonObject.hpp:166: undefined reference to MQTTtoONOFF(char*, char*)`

`collect2: error: ld returned 1 exit status`

I'm using LOLIN(WEMOS) D1 mini Pro.

**Comment by david-j84** [comment]
I am also having the same issue with a WEMOS D1 Mini, I have experimented with different versions of the ArduinoJSON library as some limited troubleshooting. I tried the 0.9.3 and the 0.9.4b beta.

**Comment by 1technophile** [comment]
Could you share your `user_config.h` or/and your env definition please?

**Comment by 1technophile** [comment]
reproduced with Arduino IDE (works with PIO)

**Comment by halk1982** [comment]
Hello 1technophile, can't thank you enough for OMG project! It's absolutely amazing from several aspects. Regarding this issue, to my understanding this will be resolved in version 0.9.9, is that correct ?

**Comment by 1technophile** [comment]
Hello;

It should be resolved from v0.9.4, are you still getting this error?

**Comment by halk1982** [comment]
It could be that I'm facing another issue then, apologies. I tried to upload to an ESP32 (Espressif) both BLE & Actuator ONOFF but failed with "Zactuator "undefined reference to `MQTTtoONOFF" error message:
 
![Screen Shot 2022-01-20 at 19 58 50](https://user-images.githubusercontent.com/40629096/150395513-1a2d876a-345c-4134-ba5f-10c75139581d.png)

Then I tried to upload only ZActuatorONOFF to an ESP8266 and again failed with these messages:

![Screen Shot 2022-01-20 at 14 45 05](https://user-images.githubusercontent.com/40629096/150350608-bd4cdb5c-0e39-4331-8772-0a6d04c2d56e.png)

P.S. I have already successfully created an ESP8266 with CC1101 and an ESP32 with BLE only.
